upgrade_testing/data/upgrade

```diff
@@ -322,6 +338,16 @@ function do_normal_upgrade() {
         upgrade_log "Proposed version found: ${version}, using it"
         do-release-upgrade -p -f DistUpgradeViewNonInteractive
     fi
+    post_upgrade_kernel_check=$(dpkg -l linux-*-$kernel)
+    if [[ "${pre_upgrade_kernel_check}" != "${post_upgrade_kernel_check}" ]]; then
+        upgrade_log "The kernel booted during the upgrade isn't running post-upgrade"
```

Comment:
It's more like: "Different packages are installed for the kernel booted during upgrade" (as in header packages disappeared or something).

Reply:
Nitpicks welcome, I'll amend